presto-tablestore/src/main/java/com/facebook/presto/tablestore/TablestoreMetadata.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    @Override
+    public List<String> listSchemaNames(ConnectorSession session)
+    {
+        return facade.listSchemaNames(session);
+    }
+
+    @Override
+    public List<SchemaTableName> listTables(ConnectorSession session, @Nullable String schemaNameOrNull)
+    {
+        return facade.listTables(session, schemaNameOrNull);
+    }
+
+    @Override
+    public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName stn)
+    {
+        return facade.getTableHandle(session, stn);
+    }
+
+    @Override
+    public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table,
+            Constraint<ColumnHandle> constraint,
+            Optional<Set<ColumnHandle>> desiredColumns)
+    {
+        TablestoreTableHandle handle = checkType(table, TablestoreTableHandle.class);
+        TupleDomain<ColumnHandle> tupleDomain = constraint.getSummary();
+        TablestoreTableLayoutHandle tlh = new TablestoreTableLayoutHandle(handle, tupleDomain);
+        ConnectorTableLayout tl = getTableLayout(session, tlh);
+        return ImmutableList.of(new ConnectorTableLayoutResult(tl, tupleDomain));
+    }
+
+    @Override
+    public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTableLayoutHandle layoutHandle)
+    {
+        TablestoreTableLayoutHandle h = checkType(layoutHandle, TablestoreTableLayoutHandle.class);
+        return new ConnectorTableLayout(h);
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
+    {
+        TablestoreTableHandle oth = checkType(table, TablestoreTableHandle.class);
+
+        List<ColumnMetadata> metadataList = oth.getOrderedColumnHandles().stream()
+                .map(TablestoreColumnHandle::getColumnMetadata)
+                .collect(Collectors.toList());
+
+        return new TablestoreConnectorTableMetadata(oth, metadataList);
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return checkType(tableHandle, TablestoreTableHandle.class).getColumnHandleMap();
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle th, ColumnHandle ch)
+    {
+        return checkType(ch, TablestoreColumnHandle.class).getColumnMetadata();
+    }
+
+    @Override
+    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
+        List<SchemaTableName> list = listTables(session, prefix.getSchemaName());
+        for (SchemaTableName stn : list) {
+            String qtn = prefix.getTableName();
+            if (qtn != null && !qtn.equalsIgnoreCase(stn.getTableName())) { //如果tableName==null，就匹配全部；否则按tableName过滤
```

Comment:
Could you remove the comment or translate it into English?

Reply:
done